mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -209,7 +209,7 @@ public void setInMailboxes(MessageId messageId, Collection<MailboxId> targetMail
 
     private List<MailboxMessage> findRelatedMailboxMessages(MessageId messageId, MailboxSession mailboxSession) throws MailboxException {
         MessageIdMapper messageIdMapper = mailboxSessionMapperFactory.getMessageIdMapper(mailboxSession);
-        return messageIdMapper.find(ImmutableList.of(messageId), MessageMapper.FetchType.Metadata)
+        return messageIdMapper.find(ImmutableList.of(messageId), MessageMapper.FetchType.Body)
```

Comment:
Ok I think this will need some discussion. 
I guess this is really bad for performance and also I guess "Body" is also not enough because we would also need to fetch attachment data.
If we do not want to do that, we will need to do a lot of heavy code in the ES backend to retrieve content related field from the old message to the new message.
But this is going to be ugly.
We could also add a event about "message metadata modification" and update the message instead of replacing it in ES

Reply:
It shows that having the same object containing different sets of data is evil : it creates temporal coupling (depending on the life of a given object, it does not always have the same capabilities).
We won't change it right now but we'll end up changing that fact.

Reply:
Should be "Full" or you take the risk of not loading the headers BTW